okta-hosted-login/src/Navbar.jsx

```diff
@@ -13,13 +13,72 @@
 import { useOktaAuth } from '@okta/okta-react';
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { Container, Icon, Image, Menu } from 'semantic-ui-react';
+import { Container, Icon, Image, Menu, Modal } from 'semantic-ui-react';
+import config from './config';
 
 const Navbar = () => {
+  const [errorModalOpen, setErrorModalOpen] = React.useState(false);
   const { authState, oktaAuth } = useOktaAuth();
 
+  // Note: Can't distinguish CORS error from other network errors
+  const isCorsError = (err) => (err.name === 'AuthApiError' && !err.errorCode && err.xhr.message === 'Failed to fetch');
+
+  // Build URL to "Trusted Origins" page in Admin panel
+  const { issuer } = config.oidc;
+  const baseUrl = issuer.split('/oauth2')[0];
+  const hostParts = new URL(baseUrl).host.split('.');
+  hostParts[0] += '-admin';
+  const adminHost = hostParts.join('.');
+  const corsAdminUrl = `https://${adminHost}/admin/access/api/trusted_origins`;
+
+  // URL to guide for enabling CORS
+  const guideUrl = 'https://developer.okta.com/docs/guides/enable-cors/granting-cors/';
+
   const login = async () => oktaAuth.signInWithRedirect();
```

Comment:
Can a CORS error also occur on signIn? (for example, requesting well-known endpoint)

Reply:
It can occur in `custom-login` sample where sign-in widget is used. 
Request to `/api/v1/authn` will fail with CORS error.
Widget will show `There was an unexpected internal error. Please try again.`
https://github.com/okta/okta-signin-widget/blob/aab4076b57a298f343f2b71cf93b5c7c939a934c/src/util/Util.js#L68

In `okta-hosted-login` sample login works fine